pkgs/build-support/make-startupitem/default.nix

```diff
@@ -22,6 +23,9 @@ stdenv.mkDerivation {
     mkdir -p $out/etc/xdg/autostart
     target=${name}.desktop
     cp ${package}/share/applications/${srcPrefix}${name}.desktop $target
+    ${lib.optionalString (extraArgs != []) ''
+      sed -i '/Exec=.*/ s/$/ ${lib.escapeRegex (lib.escapeShellArgs extraArgs)}/' $target
```

Comment:
I just know some applications put all positional arguements at the end so the field codes will often need to stay at the end of commands; perhaps we should have a `extraBeforeArgs` and `extraAfterArgs`?

sed has extended regex support `-r` so we can capture the command and existing args (including field codes) as distinct groups. 
```
# Group 1 becomes the Exec=
# Group 2 becomes the base executable command or path.
# Group 3 becomes the existing arguments.
sed -r "s/(Exec=)([^\s]*)\s(.*)/\1\2 \3/" myapp.desktop
```

Reply:
Thank you for the suggestion, I was able to get it working based on this.